### PR TITLE
[RFC] Rename `context.lists` to `context.query` and `context.db.lists` to `context.db`

### PR DESCRIPTION
**Pull request by timleslie**
(context provided by @JedWatson, this was my idea but @timleslie did the PR)

Based on my experience using and teaching the new Lists and DB APIs to developers picking up Keystone 6, they're both important and unique APIs that you really need to understand how to use well, but it's tricky to explain.

Once you get the mental model you're good but getting there isn't super intuitive. The hope is that this change will improve that for developers starting Keystone 6 or picking up Keystone project codebases in the future.

---

### TL;DR:

> If you want to query fields, use the **Query API**. It's functionally the same as the GraphQL API, available in a server-side context. If you want to retrieve data stored in the database, use the **DB API**. Input arguments, Access control and Hooks work identically in both APIs.

#### Query API

This lets you run queries and mutations against lists in the Keystone schema, as a shortcut for executing full GraphQL queries yourself. You provide the same input arguments and query the same output fields as in the GraphQL API (behind the scenes this is just sugar for executing queries against the internal schema)

Because it resolves return data through the GraphQL Schema you can query virtual fields, return deeply nested relational data, etc.

#### Database API

This lets you run the same queries and mutations against lists in the Keystone schema that the Query API does, but will return the backing data for each item directly from the database without going through the GraphQL field resolvers.

The only difference is in the return type; the input arguments (for where / data / etc) as well as access control and hooks functionality is identical between both APIs.

The primary use for this API is if you want to return a Keystone item type from a custom query or mutation in your GraphQL extensions.

Additionally, you get all the data stored in the database for the item (equivalent to `select * from {table}`) so you can access some data that fields do not make accessible through their GraphQL resolvers, like password hashes which you would need to query to compare with a candidate when validating a signing attempt (but are never available through the GraphQL API)

### CODE REVIEW

**Comment by JedWatson** [comment]
@timleslie we could alias backwards compatibility for this pretty easily, might ease the upgrade process

(deprecate the TS types, add a single (per-process) warning log when first accessed using a getter)

then remove the aliases when we actually ship to `@keystone-6`

otoh we haven't done a lot of this for other breaking api changes recently so and I'm not sure it's been a problem

**Review by emmatown — CHANGES_REQUESTED** [review]
Just want to explicitly block this for now, gonna raise some questions in slack

File: docs/components/docs/Navigation.tsx
Comment (mitchellhamilton) [inline docs/components/docs/Navigation.tsx:195]:
```suggestion
        <NavItem href="/docs/apis/query">Query API</NavItem>
```

**Comment by gautamsi** [comment]
Is this slated for release this week?

File: docs/pages/releases/2021-04-20.mdx
Comment (mitchellhamilton) [inline docs/pages/releases/2021-04-20.mdx:58]: Can you revert these changes? Having a page on a release be inaccurate about what it is true in that release is not good imo

Comment (timleslie) [reply docs/pages/releases/2021-04-20.mdx:58]: Oh, good catch, they weren't meant to go in 👍

**Comment by timleslie** [comment]
> Is this slated for release this week?

It's in the final phase of review and will be out in the next release.

File: packages/keystone/src/___internal-do-not-use-will-break-in-patch/node-api.ts
Comment (mitchellhamilton) [inline packages/keystone/src/___internal-do-not-use-will-break-in-patch/node-api.ts:6]: `createListsAPI` -> `createQueryAPI` or something + the thing we export from the `.keystone/api` thing should be renamed too, no?

File: packages/keystone/src/artifacts.ts
Comment (mitchellhamilton) [inline packages/keystone/src/artifacts.ts:129]: And should mention this in a changeset
```suggestion
export const query = createQueryAPI(keystoneConfig, PrismaClient);
```

**Comment by gautamsi** [comment]
I thought this is release week, but seems like it is next week :)

**Review by emmatown — APPROVED** [review]
Looks like some snapshots need updating but other than that, LGTM

**Pull request merged**
